### ISSUE TITLE
Choose GOROOT befopre GOPATH when selecting libraries

### ISSUE THREAD

**Issue by mxplusb** [go-lang-plugin-org/go-lang-idea-plugin#2839]
- Plugin version (or commit hash): 0.13.1914
- IDE name and version: PyCharm 2016.3
- Java version: Built-in
- OS name and version: Windows 10 x64 

- What are you trying to do?

Call a standard library and have the plugin autodetect which to import. In this case, I typed `runtime.GOOS` and let the plugin add the import.

- What would you expect to happen?

The plugin would add `"runtime"` from the GOROOT standard library, not my GOPATH.

- What happens?

Because I have the core Go language (and lots of it's ecosystem) imported into my GOPATH (for language development), I can recursively import my local Go implementation. So when I typed `runtime.GOOS`, the plugin defaulted and imported `"github.com/golang/go/src/pkg/runtime"`. I didn't notice it (off screen import) first, and when I went to compile it, it threw [this error on Windows](https://cloud.githubusercontent.com/assets/7537841/20871677/05c288a6-ba4d-11e6-8a95-43ea05c2a135.png):

```shell
>go build
utils\log.go:13:2: C source files not allowed when not using cgo or SWIG: alg.c atomic_amd64.c callback_windows_amd64.c
cgocall.c chan.c closure_amd64.c complex.c cpuprof.c float.c hashmap.c iface.c lock_sema.c mcache.c mcentral.c mem_windo
ws.c mfinal.c mfixalloc.c mgc0.c mheap.c msize.c print.c proc.c rune.c runtime.c signal_windows_amd64.c slice.c symtab.c
 thread_windows.c traceback_x86.c
```

**Comment by dlsniper** [comment]
@zolotov the order of priority should be:
- vendor/ if present
- GOROOT 
- GOPATH folders (since they can be multiple values)

**Comment by zolotov** [comment]
@dlsniper no, it shouldn't. It is the order of roots for import path resolving. It has nothing to do with an order of import paths for qualifier resolving.

@mxplusb plugin doesn't add any import silently, it doesn't even try to autodetect which to import. In case you have several package with the same name, you'll see following popup when you try to auto-import `runtime`. Notice that SDK's one is the first in the list.

<img width="913" alt="screenshot 2016-12-05 13 41 03" src="https://cloud.githubusercontent.com/assets/140920/20882133/831e0890-baf0-11e6-9cbe-8ad02eb9da10.png">

So either you have provided not exactly steps to recreate the issue or you just have no the `runtime` package attached to the project. Please describe steps to recreate the issue more accurate and attach `.idea` directory of you project.

**Comment by dlsniper** [comment]
@zolotov I've meant as according to how the go tool will search when compiling, if it helps for any priority weighting or as such.
